frontend/src/lib/components/ChildrenGallery.svelte

```diff
@@ -1,89 +1,159 @@
+<svelte:options runes={true} />
+
 <script lang="ts">
+import { getChildImage, getChildren } from "$lib/client/services.gen";
 import CardDisplay from "$lib/components/DataDisplay/CardDisplay.svelte";
 import GalleryDisplay from "$lib/components/DataDisplay/GalleryDisplay.svelte";
-import Breadcrumbs from "$lib/components/Navigation/Breadcrumbs.svelte";
-import { type ChildData, children } from "$lib/stores/childrenStore";
-
-import { createStyle, init } from "$lib/components/ChildrenGallery";
+import { currentChild } from "$lib/stores/childrenStore";
+import { activeTabChildren } from "$lib/stores/componentStore";
 import { Heading } from "flowbite-svelte";
-import { onDestroy, onMount } from "svelte";
-
-let data: ChildData[] = [];
+import { _ } from "svelte-i18n";
+import AlertMessage from "./AlertMessage.svelte";
 
-export let breadcrumbdata: any[] | null = null;
+async function setup(): Promise<any> {
+	const children = await getChildren();
 
-function searchName(data: any[], key: string): any[] {
-	if (key === "") {
-		return data;
+	if (children.error) {
+		console.log("Error when retrieving child data");
+		showAlert = true;
+		alertMessage =
+			$_("childData.alertMessageRetrieving") + children.error.detail;
 	} else {
-		const res = data.filter((item) => {
-			return item.header.toLowerCase().includes(key.toLowerCase());
-		});
-		return res;
+		const childrenData = await Promise.all(
+			children.data.map(async (child) => {
+				let image = null;
+				if (child.has_image) {
+					const childImageResponse = await getChildImage({
+						path: { child_id: child.id },
+					});
+					if (childImageResponse.error) {
+						console.log("Error when retrieving child image");
+						showAlert = true;
+						alertMessage =
+							$_("childData.alertMessageImage") +
+							childImageResponse.error.detail;
+					} else {
+						const reader = new FileReader();
+						reader.readAsDataURL(childImageResponse.data);
+						image = await new Promise((resolve) => {
+							reader.onloadend = () => resolve(reader.result as string);
+						});
+					}
+				}
+				return {
+					header: child.name,
+					image,
+					events: {
+						onclick: () => {
+							currentChild.set(child.id);
+							activeTabChildren.set("childrenRegistration");
+						},
+					},
+				};
+			}),
+		);
+
+		// add the 'new child' card as the first element
+		data = [
+			{
+				header: $_("childData.newChildHeading"),
+				summary: $_("childData.newChildHeadingLong"),
+				events: {
+					onclick: async () => {
+						currentChild.set(null);
+						activeTabChildren.set("childrenRegistration");
+					},
+				},
+				image: null,
+			},
+			...childrenData,
+		];
 	}
+	return data;
 }
 
-function searchRemarks(data: any[], key: string): any[] {
+function createStyle(data: any[]) {
```

Comment:
How about if GalleryDisplay took a list of items, where each item has
```js
{data = ..., component = ..., componentProps = ...}
```
rather than taking a separate list for each of these?
It seems (at least to me) that would reduce the need for this extra logic / data manipulation?

Reply:
yes, this is my intention, but I didn´t want to do this here as it will result in refactoring in at least 3 (frontpage, childrengallery, milestones) different places  (everywhere gallery is used)

Reply:
issue: #151